
<!DOCTYPE html>
<html lang="cn"
>
<head>
    <title>clinvar数据库的使用 - Café de Lyon</title>
    <!-- Using the latest rendering mode for IE -->
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta property="wb:webmaster" content="3cba1a80be39e3a2" />



<link rel="canonical" href="http://www.lyon0804.com/clinvarshu-ju-ku-de-shi-yong.html">

        <meta name="author" content="lyon" />
        <meta name="keywords" content="bioinformatics,clinvar,ANNOVAR" />
        <meta name="description" content="clinvar数据库 clinvar数据库记录基因组变异和人类健康的关系, 在遗传变异与疾病的研究中非常重要. 数据库获取 该数据库可以FTP站点下载到. 该数据库大约每月更新一次, 以保证数据实时性. 数据库构成 在下载到的clinvar_20160705.vcf(或者其他日期版本)中, 是在dbsnp的基础上进行注释, 实际内容在INFO中体现. 由于dbsnp中一条记录可能包含多种基因型的变异, 显然, 不同的基因型表现出的疾病关系很可能是不同的, 所以在下载的vcf文件中, vcf文件条目和实际的clinvar记录条目是一对多的关系. 意识到这一点非常重要, 由于很多工具并不会针对clinvar进行特殊处理, 忽略这个特性很有可能会得到前后矛盾的注释结果. ANNOVAR的问题 在ANNOVAR中, 作者提供了切分和向左标准化的clinvar文件, 但是在转换的同时CLINHGVS信息丢失, 所以注释的结果中并不能准确表示clinvar数据库的原始表达. 如: chr7:150648198这个位点, A&gt;G表现为良性, 而A&gt;T则表现为致病, 在原始文件中的记录是这样的: 7 150648198 rs1137617 A C,G,T . . RS=1137617;RSPOS=150648198;RV;dbSNPBuildID ..." />

        <meta property="og:site_name" content="Café de Lyon" />
        <meta property="og:type" content="article"/>
        <meta property="og:title" content="clinvar数据库的使用"/>
        <meta property="og:url" content="http://www.lyon0804.com/clinvarshu-ju-ku-de-shi-yong.html"/>
        <meta property="og:description" content="clinvar数据库 clinvar数据库记录基因组变异和人类健康的关系, 在遗传变异与疾病的研究中非常重要. 数据库获取 该数据库可以FTP站点下载到. 该数据库大约每月更新一次, 以保证数据实时性. 数据库构成 在下载到的clinvar_20160705.vcf(或者其他日期版本)中, 是在dbsnp的基础上进行注释, 实际内容在INFO中体现. 由于dbsnp中一条记录可能包含多种基因型的变异, 显然, 不同的基因型表现出的疾病关系很可能是不同的, 所以在下载的vcf文件中, vcf文件条目和实际的clinvar记录条目是一对多的关系. 意识到这一点非常重要, 由于很多工具并不会针对clinvar进行特殊处理, 忽略这个特性很有可能会得到前后矛盾的注释结果. ANNOVAR的问题 在ANNOVAR中, 作者提供了切分和向左标准化的clinvar文件, 但是在转换的同时CLINHGVS信息丢失, 所以注释的结果中并不能准确表示clinvar数据库的原始表达. 如: chr7:150648198这个位点, A&gt;G表现为良性, 而A&gt;T则表现为致病, 在原始文件中的记录是这样的: 7 150648198 rs1137617 A C,G,T . . RS=1137617;RSPOS=150648198;RV;dbSNPBuildID ..."/>
        <meta property="article:published_time" content="2016-07-15" />
            <meta property="article:section" content="bioinformatics" />
            <meta property="article:tag" content="bioinformatics" />
            <meta property="article:tag" content="clinvar" />
            <meta property="article:tag" content="ANNOVAR" />
            <meta property="article:author" content="lyon" />


    <!-- Bootstrap -->
        <link rel="stylesheet" href="http://www.lyon0804.com/theme/css/bootstrap.min.css" type="text/css"/>
    <link href="http://www.lyon0804.com/theme/css/font-awesome.min.css" rel="stylesheet">

    <link href="http://www.lyon0804.com/theme/css/pygments/native.css" rel="stylesheet">
    <link rel="stylesheet" href="http://www.lyon0804.com/theme/css/style.css" type="text/css"/>

        <link href="http://www.lyon0804.com/feeds/all.atom.xml" type="application/atom+xml" rel="alternate"
              title="Café de Lyon ATOM Feed"/>

</head>
<body>

<div class="navbar navbar-default navbar-fixed-top" role="navigation">
	<div class="container">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-ex1-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a href="http://www.lyon0804.com/" class="navbar-brand">
<img src="http://www.lyon0804.com/static/img/logo.png" width="30px"/> Café de Lyon            </a>
        </div>
        <div class="collapse navbar-collapse navbar-ex1-collapse">
            <ul class="nav navbar-nav">
                        <li class="active">
                            <a href="http://www.lyon0804.com/category/bioinformatics.html">Bioinformatics</a>
                        </li>
                        <li >
                            <a href="http://www.lyon0804.com/category/learning.html">Learning</a>
                        </li>
                        <li >
                            <a href="http://www.lyon0804.com/category/pages.html">Pages</a>
                        </li>
                        <li >
                            <a href="http://www.lyon0804.com/category/photo.html">Photo</a>
                        </li>
            </ul>
            <ul class="nav navbar-nav navbar-right">
              <li><a href="http://www.lyon0804.com/archives.html"><i class="fa fa-th-list"></i><span class="icon-label">Archives</span></a></li>
            </ul>
        </div>
        <!-- /.navbar-collapse -->
    </div>
</div> <!-- /.navbar -->
<!-- Banner -->
<!-- End Banner -->
<div class="container">
    <div class="row">
        <div class="col-lg-12">

    <section id="content">
        <article>
            <header class="page-header">
                <h1>
                    <a href="http://www.lyon0804.com/clinvarshu-ju-ku-de-shi-yong.html"
                       rel="bookmark"
                       title="Permalink to clinvar数据库的使用">
                        clinvar数据库的使用
                    </a>
                </h1>
            </header>
            <div class="entry-content">
                <div class="panel">
                    <div class="panel-body">
<footer class="post-info">
    <span class="label label-default">Date</span>
    <span class="published">
        <i class="fa fa-calendar"></i><time datetime="2016-07-15T00:00:00+08:00"> Fri 15 July 2016</time>
    </span>



<span class="label label-default">Tags</span>
	<a href="http://www.lyon0804.com/tag/bioinformatics.html">bioinformatics</a>
        /
	<a href="http://www.lyon0804.com/tag/clinvar.html">clinvar</a>
        /
	<a href="http://www.lyon0804.com/tag/annovar.html">ANNOVAR</a>
    
</footer><!-- /.post-info -->                    </div>
                </div>
                <h2>clinvar数据库</h2>
<p>clinvar数据库记录基因组变异和人类健康的关系, 在遗传变异与疾病的研究中非常重要.</p>
<h3>数据库获取</h3>
<p>该数据库可以<a href="ftp://ftp.ncbi.nlm.nih.gov/pub/clinvar/">FTP站点</a>下载到. </p>
<p>该数据库大约每月更新一次, 以保证数据实时性.</p>
<h3>数据库构成</h3>
<p>在下载到的clinvar_20160705.vcf(或者其他日期版本)中, 是在dbsnp的基础上进行注释, 实际内容在INFO中体现. </p>
<p>由于dbsnp中一条记录可能包含多种基因型的变异, 显然, 不同的基因型表现出的疾病关系很可能是不同的, 所以在下载的vcf文件中, vcf文件条目和实际的clinvar记录条目是一对多的关系. 意识到这一点非常重要, 由于很多工具并不会针对clinvar进行特殊处理, 忽略这个特性很有可能会得到前后矛盾的注释结果.</p>
<h2>ANNOVAR的问题</h2>
<p>在ANNOVAR中, 作者提供了切分和向左标准化的clinvar文件, 但是在转换的同时CLINHGVS信息丢失, 所以注释的结果中并不能准确表示clinvar数据库的原始表达.</p>
<p>如:</p>
<p>chr7:150648198这个位点, A&gt;G表现为良性, 而A&gt;T则表现为致病, 在原始文件中的记录是这样的:</p>
<div class="highlight"><pre><span class="nt">7</span>       <span class="nt">150648198</span>       <span class="nt">rs1137617</span>       <span class="nt">A</span>       <span class="nt">C</span><span class="o">,</span><span class="nt">G</span><span class="o">,</span><span class="nt">T</span>   <span class="o">.</span>       <span class="o">.</span>       <span class="nt">RS</span><span class="o">=</span><span class="nt">1137617</span><span class="o">;</span><span class="nt">RSPOS</span><span class="o">=</span><span class="nt">150648198</span><span class="o">;</span><span class="nt">RV</span><span class="o">;</span><span class="nt">dbSNPBuildID</span><span class="o">=</span><span class="nt">86</span><span class="o">;</span><span class="nt">SSR</span><span class="o">=</span><span class="nt">0</span><span class="o">;</span><span class="nt">SAO</span><span class="o">=</span><span class="nt">1</span><span class="o">;</span><span class="nt">VP</span><span class="o">=</span><span class="nt">0x05017000030515053e110100</span><span class="o">;</span><span class="nt">GENEINFO</span><span class="o">=</span><span class="nt">KCNH2</span><span class="nd">:3757</span><span class="o">;</span><span class="nt">WGT</span><span class="o">=</span><span class="nt">1</span><span class="o">;</span><span class="nt">VC</span><span class="o">=</span><span class="nt">SNV</span><span class="o">;</span><span class="nt">PM</span><span class="o">;</span><span class="nt">TPA</span><span class="o">;</span><span class="nt">SLO</span><span class="o">;</span><span class="nt">REF</span><span class="o">;</span><span class="nt">SYN</span><span class="o">;</span><span class="nt">ASP</span><span class="o">;</span><span class="nt">VLD</span><span class="o">;</span><span class="nt">G5</span><span class="o">;</span><span class="nt">HD</span><span class="o">;</span><span class="nt">GNO</span><span class="o">;</span><span class="nt">KGPhase1</span><span class="o">;</span><span class="nt">KGPhase3</span><span class="o">;</span><span class="nt">LSD</span><span class="o">;</span><span class="nt">OM</span><span class="o">;</span><span class="nt">CLNALLE</span><span class="o">=</span><span class="nt">2</span><span class="o">,</span><span class="nt">3</span><span class="o">;</span><span class="nt">CLNHGVS</span><span class="o">=</span><span class="nt">NC_000007</span><span class="nc">.13</span><span class="nd">:g</span><span class="nc">.150648198A</span><span class="o">&gt;</span><span class="nt">G</span><span class="o">,</span><span class="nt">NC_000007</span><span class="nc">.13</span><span class="nd">:g</span><span class="nc">.150648198A</span><span class="o">&gt;</span><span class="nt">T</span><span class="o">;</span><span class="nt">CLNSRC</span><span class="o">=.,.;</span><span class="nt">CLNORIGIN</span><span class="o">=</span><span class="nt">1</span><span class="o">,</span><span class="nt">1</span><span class="o">;</span><span class="nt">CLNSRCID</span><span class="o">=.,.;</span><span class="nt">CLNSIG</span><span class="o">=</span><span class="nt">2</span><span class="o">,</span><span class="nt">5</span><span class="o">;</span><span class="nt">CLNDSDB</span><span class="o">=</span><span class="nt">MedGen</span><span class="o">,</span><span class="nt">MedGen</span><span class="o">;</span><span class="nt">CLNDSDBID</span><span class="o">=</span><span class="nt">CN169374</span><span class="o">,</span><span class="nt">CN221809</span><span class="o">;</span><span class="nt">CLNDBN</span><span class="o">=</span><span class="nt">not_specified</span><span class="o">,</span><span class="nt">not_provided</span><span class="o">;</span><span class="nt">CLNREVSTAT</span><span class="o">=</span><span class="nt">single</span><span class="o">,</span><span class="nt">single</span><span class="o">;</span><span class="nt">CLNACC</span><span class="o">=</span><span class="nt">RCV000181727</span><span class="nc">.1</span><span class="o">,</span><span class="nt">RCV000181832</span><span class="nc">.1</span><span class="o">;</span><span class="nt">CAF</span><span class="o">=</span><span class="nt">0</span><span class="nc">.2278</span><span class="o">,.,</span><span class="nt">0</span><span class="nc">.7722</span><span class="o">,.;</span><span class="nt">COMMON</span><span class="o">=</span><span class="nt">1</span>
</pre></div>


<p>在ANNOVAR转换后的clinvar中, 该记录是这样的:</p>
<div class="highlight"><pre>7       150648198       150648198       A       C       Benign\x2cPathogenic    Cardiac_arrhythmia\x2cCardiac_arrhythmia        RCV000181727.1,RCV000181832.1   MedGen:OMIM\x2cMedGen:OMIM      CN029864:115000\x2cCN029864:115000
7       150648198       150648198       A       G       Benign\x2cPathogenic    Cardiac_arrhythmia\x2cCardiac_arrhythmia        RCV000181727.1,RCV000181832.1   MedGen:OMIM\x2cMedGen:OMIM      CN029864:115000\x2cCN029864:115000
7       150648198       150648198       A       T       Benign\x2cPathogenic    Cardiac_arrhythmia\x2cCardiac_arrhythmia        RCV000181727.1,RCV000181832.1   MedGen:OMIM\x2cMedGen:OMIM      CN029864:115000\x2cCN029864:115000
</pre></div>


<p>虽然ANNOVAR对clinvar进行了切分, 将原始文件中的复合记录切分为三条, 但是在切分的过程中并没有区分CLNHGVS和其他clnvar记录的对应关系;</p>
<p>如在原始文件中, NC_000007.13:g.150648198A&gt;G对应的CLNSIG是2(Benign), 而NC_000007.13:g.150648198A&gt;T对应的CLNSIG则是5(Pathogenic), 而在ANNOVAR转换之后, 虽然拆分成了A&gt;C, A&gt;G, A&gt;T三条记录, 但是关键的clinvar信息却是简单拼接的;</p>
<p>可想而知, 使用ANNOVAR对chr7:150648198A&gt;C这条变异的结果进行注释时, 会得到致病的判断, 这显然是错误的.</p>
<h2>Clinvar的切分和标准化</h2>
<p>由于clinvar提供的vcf文件特点, 一条记录中可能包含同一个位点的不同变异, 这样并不便于后续的注释, 所以切分是必要的, 但是在切分的过程中, 需要根据CLNHGVS对应的基因型, 正确反应原始记录.</p>
<h3>根据HGVS判断变异</h3>
<p>CLNHGVS使用HGVS标准来表示一个变异, 不同的变异类型其表示方式分别举例有:</p>
<ul>
<li>snp: NC_000001.10:g.955597G&gt;T;</li>
<li>ins: NC_000001.10:g.2337967_2337968insC, 注意C插入的位置是在2337967之后;</li>
<li>del: NC_000001.10:g.2340153delA, NC_000001.10:g.6529183_6529185delTCC;</li>
<li>mnp: NC_000001.10:g.1273413_1273420delTAGGCAGGinsC, NC_000001.10:g.201334375_201334376delinsAC;</li>
<li>dup: NC_000001.10:g.949699dupG, NC_000001.10:g.3334509_3334510dupCC;</li>
<li>野生型: NC_000001.10:g.25717365C\x3d;</li>
</ul>
<p>通过HGVS可以准确定位到染色体位置, 并获取变异的相关信息;</p>
<h3>通过CLNHGVS进行切分</h3>
<p>在INFO中CLN开头的字段表示clinvar的注释信息, 由于一个记录可能对应多个变异, 所以CLN开头的字段都是逗号分隔的列表; </p>
<p>根据CLNHGVS可以获得有注释的基因型列表, 通过该列表对clinvar原始文件进行切分而不是通过ALT进行切分, 好处是可以准确表达clinvar中存在记录的变异;</p>
<p>CLN开头的字段列表是一一对应的, 所以在切分之后, 需要一一获取其中的信息;</p>
<p>如:</p>
<div class="highlight"><pre><span class="nt">7</span>       <span class="nt">150648198</span>       <span class="nt">rs1137617</span>       <span class="nt">A</span>       <span class="nt">C</span><span class="o">,</span><span class="nt">G</span><span class="o">,</span><span class="nt">T</span>   <span class="o">.</span>       <span class="o">.</span>       <span class="nt">RS</span><span class="o">=</span><span class="nt">1137617</span><span class="o">;</span><span class="nt">RSPOS</span><span class="o">=</span><span class="nt">150648198</span><span class="o">;</span><span class="nt">RV</span><span class="o">;</span><span class="nt">dbSNPBuildID</span><span class="o">=</span><span class="nt">86</span><span class="o">;</span><span class="nt">SSR</span><span class="o">=</span><span class="nt">0</span><span class="o">;</span><span class="nt">SAO</span><span class="o">=</span><span class="nt">1</span><span class="o">;</span><span class="nt">VP</span><span class="o">=</span><span class="nt">0x05017000030515053e110100</span><span class="o">;</span><span class="nt">GENEINFO</span><span class="o">=</span><span class="nt">KCNH2</span><span class="nd">:3757</span><span class="o">;</span><span class="nt">WGT</span><span class="o">=</span><span class="nt">1</span><span class="o">;</span><span class="nt">VC</span><span class="o">=</span><span class="nt">SNV</span><span class="o">;</span><span class="nt">PM</span><span class="o">;</span><span class="nt">TPA</span><span class="o">;</span><span class="nt">SLO</span><span class="o">;</span><span class="nt">REF</span><span class="o">;</span><span class="nt">SYN</span><span class="o">;</span><span class="nt">ASP</span><span class="o">;</span><span class="nt">VLD</span><span class="o">;</span><span class="nt">G5</span><span class="o">;</span><span class="nt">HD</span><span class="o">;</span><span class="nt">GNO</span><span class="o">;</span><span class="nt">KGPhase1</span><span class="o">;</span><span class="nt">KGPhase3</span><span class="o">;</span><span class="nt">LSD</span><span class="o">;</span><span class="nt">OM</span><span class="o">;</span><span class="nt">CLNALLE</span><span class="o">=</span><span class="nt">2</span><span class="o">,</span><span class="nt">3</span><span class="o">;</span><span class="nt">CLNHGVS</span><span class="o">=</span><span class="nt">NC_000007</span><span class="nc">.13</span><span class="nd">:g</span><span class="nc">.150648198A</span><span class="o">&gt;</span><span class="nt">G</span><span class="o">,</span><span class="nt">NC_000007</span><span class="nc">.13</span><span class="nd">:g</span><span class="nc">.150648198A</span><span class="o">&gt;</span><span class="nt">T</span><span class="o">;</span><span class="nt">CLNSRC</span><span class="o">=.,.;</span><span class="nt">CLNORIGIN</span><span class="o">=</span><span class="nt">1</span><span class="o">,</span><span class="nt">1</span><span class="o">;</span><span class="nt">CLNSRCID</span><span class="o">=.,.;</span><span class="nt">CLNSIG</span><span class="o">=</span><span class="nt">2</span><span class="o">,</span><span class="nt">5</span><span class="o">;</span><span class="nt">CLNDSDB</span><span class="o">=</span><span class="nt">MedGen</span><span class="o">,</span><span class="nt">MedGen</span><span class="o">;</span><span class="nt">CLNDSDBID</span><span class="o">=</span><span class="nt">CN169374</span><span class="o">,</span><span class="nt">CN221809</span><span class="o">;</span><span class="nt">CLNDBN</span><span class="o">=</span><span class="nt">not_specified</span><span class="o">,</span><span class="nt">not_provided</span><span class="o">;</span><span class="nt">CLNREVSTAT</span><span class="o">=</span><span class="nt">single</span><span class="o">,</span><span class="nt">single</span><span class="o">;</span><span class="nt">CLNACC</span><span class="o">=</span><span class="nt">RCV000181727</span><span class="nc">.1</span><span class="o">,</span><span class="nt">RCV000181832</span><span class="nc">.1</span><span class="o">;</span><span class="nt">CAF</span><span class="o">=</span><span class="nt">0</span><span class="nc">.2278</span><span class="o">,.,</span><span class="nt">0</span><span class="nc">.7722</span><span class="o">,.;</span><span class="nt">COMMON</span><span class="o">=</span><span class="nt">1</span>
</pre></div>


<p>以上记录中, 正确切分后应该为:</p>
<div class="highlight"><pre><span class="nt">chr7</span>    <span class="nt">150648198</span>       <span class="nt">rs1137617</span>       <span class="nt">A</span>       <span class="nt">G</span>       <span class="o">.</span>       <span class="o">.</span>       <span class="nt">dbSNPBuildID</span><span class="o">=</span><span class="nt">86</span><span class="o">;</span><span class="nt">RV</span><span class="o">;</span><span class="nt">RSPOS</span><span class="o">=</span><span class="nt">150648198</span><span class="o">;</span><span class="nt">KGPhase1</span><span class="o">;</span><span class="nt">SSR</span><span class="o">=</span><span class="nt">0</span><span class="o">;</span><span class="nt">ASP</span><span class="o">;</span><span class="nt">OM</span><span class="o">;</span><span class="nt">GNO</span><span class="o">;</span><span class="nt">RS</span><span class="o">=</span><span class="nt">1137617</span><span class="o">;</span><span class="nt">CLNORIGIN</span><span class="o">=</span><span class="nt">1</span><span class="o">;</span><span class="nt">CAF</span><span class="o">=</span><span class="nt">0</span><span class="nc">.2278</span><span class="o">,.,</span><span class="nt">0</span><span class="nc">.7722</span><span class="o">,.;</span><span class="nt">WGT</span><span class="o">=</span><span class="nt">1</span><span class="o">;</span><span class="nt">CLNACC</span><span class="o">=</span><span class="nt">RCV000181727</span><span class="nc">.1</span><span class="o">;</span><span class="nt">SAO</span><span class="o">=</span><span class="nt">1</span><span class="o">;</span><span class="nt">VLD</span><span class="o">;</span><span class="nt">CLNDBN</span><span class="o">=</span><span class="nt">not_specified</span><span class="o">;</span><span class="nt">G5</span><span class="o">;</span><span class="nt">PM</span><span class="o">;</span><span class="nt">CLNREVSTAT</span><span class="o">=</span><span class="nt">single</span><span class="o">;</span><span class="nt">LSD</span><span class="o">;</span><span class="nt">COMMON</span><span class="o">=</span><span class="nt">1</span><span class="o">;</span><span class="nt">CLNDSDB</span><span class="o">=</span><span class="nt">MedGen</span><span class="o">;</span><span class="nt">REF</span><span class="o">;</span><span class="nt">TPA</span><span class="o">;</span><span class="nt">VP</span><span class="o">=</span><span class="nt">0x05017000030515053e110100</span><span class="o">;</span><span class="nt">SYN</span><span class="o">;</span><span class="nt">VC</span><span class="o">=</span><span class="nt">SNV</span><span class="o">;</span><span class="nt">GENEINFO</span><span class="o">=</span><span class="nt">KCNH2</span><span class="nd">:3757</span><span class="o">;</span><span class="nt">CLNSIG</span><span class="o">=</span><span class="nt">2</span><span class="o">;</span><span class="nt">SLO</span><span class="o">;</span><span class="nt">CLNHGVS</span><span class="o">=</span><span class="nt">NC_000007</span><span class="nc">.13</span><span class="nd">:g</span><span class="nc">.150648198A</span><span class="o">&gt;</span><span class="nt">G</span><span class="o">;</span><span class="nt">CLNSRCID</span><span class="o">=.;</span><span class="nt">KGPhase3</span><span class="o">;</span><span class="nt">HD</span><span class="o">;</span><span class="nt">CLNSRC</span><span class="o">=.;</span><span class="nt">CLNDSDBID</span><span class="o">=</span><span class="nt">CN169374</span>
<span class="nt">chr7</span>    <span class="nt">150648198</span>       <span class="nt">rs1137617</span>       <span class="nt">A</span>       <span class="nt">T</span>       <span class="o">.</span>       <span class="o">.</span>       <span class="nt">dbSNPBuildID</span><span class="o">=</span><span class="nt">86</span><span class="o">;</span><span class="nt">RV</span><span class="o">;</span><span class="nt">RSPOS</span><span class="o">=</span><span class="nt">150648198</span><span class="o">;</span><span class="nt">KGPhase1</span><span class="o">;</span><span class="nt">SSR</span><span class="o">=</span><span class="nt">0</span><span class="o">;</span><span class="nt">ASP</span><span class="o">;</span><span class="nt">OM</span><span class="o">;</span><span class="nt">GNO</span><span class="o">;</span><span class="nt">RS</span><span class="o">=</span><span class="nt">1137617</span><span class="o">;</span><span class="nt">CLNORIGIN</span><span class="o">=</span><span class="nt">1</span><span class="o">;</span><span class="nt">CAF</span><span class="o">=</span><span class="nt">0</span><span class="nc">.2278</span><span class="o">,.,</span><span class="nt">0</span><span class="nc">.7722</span><span class="o">,.;</span><span class="nt">WGT</span><span class="o">=</span><span class="nt">1</span><span class="o">;</span><span class="nt">CLNACC</span><span class="o">=</span><span class="nt">RCV000181832</span><span class="nc">.1</span><span class="o">;</span><span class="nt">SAO</span><span class="o">=</span><span class="nt">1</span><span class="o">;</span><span class="nt">VLD</span><span class="o">;</span><span class="nt">CLNDBN</span><span class="o">=</span><span class="nt">not_provided</span><span class="o">;</span><span class="nt">G5</span><span class="o">;</span><span class="nt">PM</span><span class="o">;</span><span class="nt">CLNREVSTAT</span><span class="o">=</span><span class="nt">single</span><span class="o">;</span><span class="nt">LSD</span><span class="o">;</span><span class="nt">COMMON</span><span class="o">=</span><span class="nt">1</span><span class="o">;</span><span class="nt">CLNDSDB</span><span class="o">=</span><span class="nt">MedGen</span><span class="o">;</span><span class="nt">REF</span><span class="o">;</span><span class="nt">TPA</span><span class="o">;</span><span class="nt">VP</span><span class="o">=</span><span class="nt">0x05017000030515053e110100</span><span class="o">;</span><span class="nt">SYN</span><span class="o">;</span><span class="nt">VC</span><span class="o">=</span><span class="nt">SNV</span><span class="o">;</span><span class="nt">GENEINFO</span><span class="o">=</span><span class="nt">KCNH2</span><span class="nd">:3757</span><span class="o">;</span><span class="nt">CLNSIG</span><span class="o">=</span><span class="nt">5</span><span class="o">;</span><span class="nt">SLO</span><span class="o">;</span><span class="nt">CLNHGVS</span><span class="o">=</span><span class="nt">NC_000007</span><span class="nc">.13</span><span class="nd">:g</span><span class="nc">.150648198A</span><span class="o">&gt;</span><span class="nt">T</span><span class="o">;</span><span class="nt">CLNSRCID</span><span class="o">=.;</span><span class="nt">KGPhase3</span><span class="o">;</span><span class="nt">HD</span><span class="o">;</span><span class="nt">CLNSRC</span><span class="o">=.;</span><span class="nt">CLNDSDBID</span><span class="o">=</span><span class="nt">CN221809</span>
</pre></div>


<p>可见在切分之后不仅注释结果更加清晰, 还去掉了A&gt;C等不存在记录的变异;</p>
<h3>记录的标准化</h3>
<p>在CLNHGVS中, 很多的表示都并非标准化的表示, 为了方便注释, 应该在切分之后进行标准化操作(左对齐, 简约化);</p>
<p>标准化操作可以参考<a href="http://www.lyon0804.com/fan-yi-variant-normalization.html">之前的文章</a>;</p>
<h2>实现</h2>
<p>具体实现脚本可以参考本人的这篇<a href="https://gist.github.com/lozybean/9bb99e48d9d155510383ac9533741113">gist</a>;</p>
<p>其中参考基因组的读取借助了<code>samtools faidx</code>命令, 这需要安装samtools并且对参考基因组fasta建立索引.</p>
            </div>
            <!-- /.entry-content -->

    <hr/>
    <section class="comments" id="comments">
        <h2>Comments</h2>
<!--         <div class="ds-thread" data-thread-key="clinvarshu-ju-ku-de-shi-yong" data-title="clinvar数据库的使用" data-url="http://www.lyon0804.com/clinvarshu-ju-ku-de-shi-yong.html"></div> -->
		<script type="text/javascript">
     		var duoshuoQuery = {short_name:"lyon0804"};
    		(function() {
    			var ds = document.createElement('script');
    			ds.type = 'text/javascript';ds.async = true;
    			ds.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') + '//static.duoshuo.com/embed.js';
    			ds.charset = 'UTF-8';
    			(document.getElementsByTagName('head')[0]
    			 || document.getElementsByTagName('body')[0]).appendChild(ds);
    		})();
    	</script>
        <script type="text/javascript">
            function toggleDuoshuoComments(container, id, title, url){
                if(jQuery(container).has("div").length>0){
                    jQuery(container).empty();
                    return;
                }
                var el = document.createElement('div');
                el.setAttribute('class','ds-thread');
                el.setAttribute('data-thread-key', id);
                el.setAttribute('data-title',title);
                el.setAttribute('data-url', url);
                DUOSHUO.EmbedThread(el);
                jQuery(container).append(el);
            }
        </script>
        <a href="javascript:void(0);" onclick="toggleDuoshuoComments('#comment-box', 'clinvarshu-ju-ku-de-shi-yong', 'clinvar数据库的使用' , 'http://www.lyon0804.com/clinvarshu-ju-ku-de-shi-yong.html');">
        查看评论</a>
        <div id="comment-box" ></div>
        <hr/>
    	<noscript>Please enable JavaScript to view the <a href="http://duoshuo.com/">comments powered by
        Duoshuo.</a></noscript>
    	<a href="http://duoshuo.com" class="dsq-brlink">comments powered by <span>Duoshuo</span></a>

    </section>
        </article>
    </section>

        </div>
    </div>
</div>
<footer>
   <div class="container">
      <hr>
      <div class="row">
            <!-- <div class="list-group-item"> -->
            <!-- <h4> -->
            <!-- <i class="fa fa-home fa-lg"></i> -->
            <!-- <span class="icon-label">Social</span></h4> -->
              <ul class="list-group" id="social">
                   <!-- <li class="list-group-item"> -->
                   <div class="social-item">
                   <a href="http://github.com/lozybean">
                     <i class="fa fa-github-square fa-lg"></i><!--  github -->
                   </a>
                   </div>
                   <!-- </li> -->
                   <!-- <li class="list-group-item"> -->
                   <div class="social-item">
                   <a href="http://weibo.com/lyon0804">
                     <i class="fa fa-weibo fa-lg"></i><!--  weibo -->
                   </a>
                   </div>
                   <!-- </li> -->
              </ul>
            <!-- </div> -->
         <div class="col-xs-10">&copy; 2016 lyon
            &middot; Powered by <a href="https://github.com/DandyDev/pelican-bootstrap3" target="_blank">pelican-bootstrap3(modified by lyon)</a>,
            <a href="http://docs.getpelican.com/" target="_blank">Pelican</a>,
            <a href="http://getbootstrap.com" target="_blank">Bootstrap</a>         </div>
         <div class="col-xs-2" id="back-top">
            <!-- <p class="pull-right"><i class="fa fa-arrow-up"></i>  -->
            <a href="#top" title="回到顶部"></a>
            </p>
         </div>
      </div>
   </div>
</footer>
<script src="http://www.lyon0804.com/theme/js/jquery.min.js"></script>

<!-- Include all compiled plugins (below), or include individual files as needed -->
<script src="http://www.lyon0804.com/theme/js/bootstrap.min.js"></script>

<!-- Enable responsive features in IE8 with Respond.js (https://github.com/scottjehl/Respond) -->
<script src="http://www.lyon0804.com/theme/js/respond.min.js"></script>


</body>
</html>