
<!DOCTYPE html>
<html lang="cn"
>
<head>
    <title>第一个生物信息app - Café de Lyon</title>
    <!-- Using the latest rendering mode for IE -->
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta property="wb:webmaster" content="3cba1a80be39e3a2" />



<link rel="canonical" href="http://www.lyon0804.com/di-yi-ge-sheng-wu-xin-xi-app.html">

        <meta name="author" content="lyon" />
        <meta name="keywords" content="pyqt,bioinformatics,mongodb" />
        <meta name="description" content="引子 本文将介绍一个桌面应用的创建全过程, 展示生物信息除了拒人于千里之外命令行界面之外, 对无IT经验的用户更加友好的展示方式; 设计环境以及技术: CentOS6.7(服务端) Windows10(客户端) Python3.5 PyQt5.6 Mongodb3 pymongo3 pyinstaller3.2 最终完成的功能: 根据基因名称, 检索所有的转录本编号, 以及对应的外显子的长度和数量 数据获取 首先从NCBIFTP站点下载准备文件:ref_GRCh37.p5_top_level.gff3.gz, 参考之前的文章的方式来下载该文件到服务器. 该文件中包含人类基因组GRCh37版本的注释信息, 利用该信息可以检索到每个基因、转录本、外显子、CDS等情况; 预览该文件可以发现, 除了注释行之外, 每行开头有NC_、NM_等标记, 这些标记是NCBI的genbank标记, 其具体意义为: AC_(Genomic): Complete genomic ..." />

        <meta property="og:site_name" content="Café de Lyon" />
        <meta property="og:type" content="article"/>
        <meta property="og:title" content="第一个生物信息app"/>
        <meta property="og:url" content="http://www.lyon0804.com/di-yi-ge-sheng-wu-xin-xi-app.html"/>
        <meta property="og:description" content="引子 本文将介绍一个桌面应用的创建全过程, 展示生物信息除了拒人于千里之外命令行界面之外, 对无IT经验的用户更加友好的展示方式; 设计环境以及技术: CentOS6.7(服务端) Windows10(客户端) Python3.5 PyQt5.6 Mongodb3 pymongo3 pyinstaller3.2 最终完成的功能: 根据基因名称, 检索所有的转录本编号, 以及对应的外显子的长度和数量 数据获取 首先从NCBIFTP站点下载准备文件:ref_GRCh37.p5_top_level.gff3.gz, 参考之前的文章的方式来下载该文件到服务器. 该文件中包含人类基因组GRCh37版本的注释信息, 利用该信息可以检索到每个基因、转录本、外显子、CDS等情况; 预览该文件可以发现, 除了注释行之外, 每行开头有NC_、NM_等标记, 这些标记是NCBI的genbank标记, 其具体意义为: AC_(Genomic): Complete genomic ..."/>
        <meta property="article:published_time" content="2016-05-30" />
            <meta property="article:section" content="bioinformatics" />
            <meta property="article:tag" content="pyqt" />
            <meta property="article:tag" content="bioinformatics" />
            <meta property="article:tag" content="mongodb" />
            <meta property="article:author" content="lyon" />


    <!-- Bootstrap -->
        <link rel="stylesheet" href="http://www.lyon0804.com/theme/css/bootstrap.min.css" type="text/css"/>
    <link href="http://www.lyon0804.com/theme/css/font-awesome.min.css" rel="stylesheet">

    <link href="http://www.lyon0804.com/theme/css/pygments/native.css" rel="stylesheet">
    <link rel="stylesheet" href="http://www.lyon0804.com/theme/css/style.css" type="text/css"/>

        <link href="http://www.lyon0804.com/feeds/all.atom.xml" type="application/atom+xml" rel="alternate"
              title="Café de Lyon ATOM Feed"/>

</head>
<body>

<div class="navbar navbar-default navbar-fixed-top" role="navigation">
	<div class="container">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-ex1-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a href="http://www.lyon0804.com/" class="navbar-brand">
<img src="http://www.lyon0804.com/static/img/logo.png" width="30px"/> Café de Lyon            </a>
        </div>
        <div class="collapse navbar-collapse navbar-ex1-collapse">
            <ul class="nav navbar-nav">
                         <li><a href="http://www.lyon0804.com/pages/shi-yong-wkhtmltopdf.html">
                             使用wkhtmltopdf
                          </a></li>
                        <li class="active">
                            <a href="http://www.lyon0804.com/category/bioinformatics.html">Bioinformatics</a>
                        </li>
                        <li >
                            <a href="http://www.lyon0804.com/category/learning.html">Learning</a>
                        </li>
                        <li >
                            <a href="http://www.lyon0804.com/category/pages.html">Pages</a>
                        </li>
                        <li >
                            <a href="http://www.lyon0804.com/category/photo.html">Photo</a>
                        </li>
            </ul>
            <ul class="nav navbar-nav navbar-right">
              <li><a href="http://www.lyon0804.com/archives.html"><i class="fa fa-th-list"></i><span class="icon-label">Archives</span></a></li>
            </ul>
        </div>
        <!-- /.navbar-collapse -->
    </div>
</div> <!-- /.navbar -->
<!-- Banner -->
<!-- End Banner -->
<div class="container">
    <div class="row">
        <div class="col-lg-12">

    <section id="content">
        <article>
            <header class="page-header">
                <h1>
                    <a href="http://www.lyon0804.com/di-yi-ge-sheng-wu-xin-xi-app.html"
                       rel="bookmark"
                       title="Permalink to 第一个生物信息app">
                        第一个生物信息app
                    </a>
                </h1>
            </header>
            <div class="entry-content">
                <div class="panel">
                    <div class="panel-body">
<footer class="post-info">
    <span class="label label-default">Date</span>
    <span class="published">
        <i class="fa fa-calendar"></i><time datetime="2016-05-30T00:00:00+08:00"> Mon 30 May 2016</time>
    </span>



<span class="label label-default">Tags</span>
	<a href="http://www.lyon0804.com/tag/pyqt.html">pyqt</a>
        /
	<a href="http://www.lyon0804.com/tag/bioinformatics.html">bioinformatics</a>
        /
	<a href="http://www.lyon0804.com/tag/mongodb.html">mongodb</a>
    
</footer><!-- /.post-info -->                    </div>
                </div>
                <h2>引子</h2>
<p>本文将介绍一个桌面应用的创建全过程, 展示生物信息除了拒人于千里之外命令行界面之外, 对无IT经验的用户更加友好的展示方式;</p>
<p>设计环境以及技术:</p>
<ul>
<li>CentOS6.7(服务端)</li>
<li>Windows10(客户端)</li>
<li>Python3.5</li>
<li>PyQt5.6</li>
<li>Mongodb3</li>
<li>pymongo3</li>
<li>pyinstaller3.2</li>
</ul>
<p>最终完成的功能:</p>
<ul>
<li>根据基因名称, 检索所有的转录本编号, 以及对应的外显子的长度和数量</li>
</ul>
<h2>数据获取</h2>
<p>首先从<a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/H_sapiens/ARCHIVE/BUILD.37.3/GFF/ref_GRCh37.p5_top_level.gff3.gz">NCBIFTP站点</a>下载准备文件:<em>ref_GRCh37.p5_top_level.gff3.gz</em>, 参考<a href="http://www.lyon0804.com/linuxwen-jian-xia-zai-fang-shi-hui-zong.html">之前的文章</a>的方式来下载该文件到服务器.</p>
<p>该文件中包含人类基因组GRCh37版本的注释信息, 利用该信息可以检索到每个基因、转录本、外显子、CDS等情况;</p>
<p>预览该文件可以发现, 除了注释行之外, 每行开头有<strong>NC_</strong>、<strong>NM_</strong>等标记, 这些标记是NCBI的genbank标记, 其具体意义为:</p>
<ul>
<li><code>AC_</code>(Genomic): Complete genomic molecule, usually alternate assembly</li>
<li><code>NC_</code>(Genomic): Complete genomic molecule, usually reference assembly</li>
<li><code>NG_</code>(Genomic): Incomplete genomic region</li>
<li><code>NT_</code>(Genomic): Contig or scaffold, clone-based or WGS</li>
<li><code>NW_</code>(Genomic): Contig or scaffold, primarily WGS</li>
<li><code>NS_</code>(Genomic): Environmental sequence</li>
<li><code>NZ_</code>(Genomic): Unfinished WGS</li>
<li><code>NM_</code>(mRNA): normal used</li>
<li><code>NR_</code>(RNA): normal used</li>
<li><code>XM_</code>(mRNA): Predicted model</li>
<li><code>XR_</code>(RNA): Predicted model</li>
<li><code>AP_</code>(Protein): Annotated on AC_ alternate assembly</li>
<li><code>NP_</code>(Protein): Associated with an NM_ or NC_ accession</li>
<li><code>YP_</code>(Protein)</li>
<li><code>XP_</code>(Protein): Predicted model, associated with an XM_ accession</li>
<li><code>ZP_</code>(Protein): Predicted model, annotated on NZ_ genomic records</li>
</ul>
<p>其中<strong>NC_</strong>开头的序列通常作为标准参考, NC_000001到NC000024分别表示24条染色体(包含XY性染色体), 是我们的重点关注对象;</p>
<h2>创建数据库</h2>
<p>为了方便检索信息, 需要选择一款数据库软件, 这里选择Mongodb来完成这次设计, mongodb的安装不再赘述;</p>
<p>在完成Mongodb的安装之后, 别忙着启动数据库, 先编辑mongodb.conf配置, 确保修改以下内容:</p>
<div class="highlight"><pre>bind_ip = 0.0.0.0
port = 27017
fork = true
auth = true
</pre></div>


<p>然后使用<code>mongod -f mongodb.conf</code>启动数据库服务;</p>
<p>并且在本地启动一个mongo shell: <code>mongo 127.0.0.1:27017</code>以便配置用户信息:</p>
<div class="highlight"><pre><span class="err">#</span> <span class="nx">添加管理员用户</span>
<span class="nx">use</span> <span class="nx">admin</span>
<span class="nx">db</span><span class="p">.</span><span class="nx">addUser</span><span class="p">(</span><span class="s1">&#39;root&#39;</span><span class="p">,</span> <span class="s1">&#39;123&#39;</span><span class="p">)</span>

<span class="err">#</span> <span class="nx">创建reference数据库</span>
<span class="nx">use</span> <span class="nx">reference</span>
<span class="err">#</span> <span class="nx">添加只读用户</span>
<span class="nx">db</span><span class="p">.</span><span class="nx">addUser</span><span class="p">(</span><span class="s1">&#39;user&#39;</span><span class="p">,</span> <span class="s1">&#39;123&#39;</span><span class="p">,</span> <span class="kc">true</span><span class="p">)</span>
</pre></div>


<p>以上配置为了客户端和服务器的权限分离, 防止客户端修改数据库;</p>
<p>利用一个小脚本, 将我们需要的信息添加到数据库中:</p>
<div class="highlight"><pre><span class="kn">import</span> <span class="nn">gzip</span>
<span class="kn">from</span> <span class="nn">pymongo</span> <span class="kn">import</span> <span class="n">MongoClient</span>

<span class="n">client</span> <span class="o">=</span> <span class="n">MongoClient</span><span class="p">(</span><span class="s">&#39;127.0.0.1&#39;</span><span class="p">,</span> <span class="mi">27017</span><span class="p">)</span>
<span class="n">db</span> <span class="o">=</span> <span class="n">client</span><span class="o">.</span><span class="n">reference</span>
<span class="n">db</span><span class="o">.</span><span class="n">authenticate</span><span class="p">(</span><span class="s">&#39;root&#39;</span><span class="p">,</span> <span class="s">&#39;123&#39;</span><span class="p">)</span>

<span class="k">with</span> <span class="n">gzip</span><span class="o">.</span><span class="n">open</span><span class="p">(</span><span class="s">&#39;ref_GRCh37.p5_top_level.gff3.gz&#39;</span><span class="p">,</span> <span class="s">&#39;rt&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">fp</span><span class="p">:</span>
    <span class="k">for</span> <span class="n">line</span> <span class="ow">in</span> <span class="n">fp</span><span class="p">:</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="n">line</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s">&#39;NC&#39;</span><span class="p">):</span>
            <span class="k">continue</span>
        <span class="p">(</span><span class="n">assembly</span><span class="p">,</span> <span class="n">_</span><span class="p">,</span> <span class="n">type_</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span><span class="p">,</span> <span class="n">_</span><span class="p">,</span>
         <span class="n">strand</span><span class="p">,</span> <span class="n">_</span><span class="p">,</span> <span class="n">items</span><span class="p">)</span> <span class="o">=</span> <span class="n">line</span><span class="o">.</span><span class="n">rstrip</span><span class="p">()</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s">&#39;</span><span class="se">\t</span><span class="s">&#39;</span><span class="p">)</span>
        <span class="n">chr_</span> <span class="o">=</span> <span class="s">&#39;chr</span><span class="si">%d</span><span class="s">&#39;</span> <span class="o">%</span> <span class="nb">int</span><span class="p">(</span><span class="n">assembly</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s">&#39;.&#39;</span><span class="p">)[</span><span class="mi">0</span><span class="p">][</span><span class="mi">3</span><span class="p">:])</span>
        <span class="k">if</span> <span class="n">type_</span> <span class="o">==</span> <span class="s">&#39;exon&#39;</span> <span class="ow">or</span> <span class="n">type_</span> <span class="o">==</span> <span class="s">&#39;mRNA&#39;</span><span class="p">:</span>
            <span class="n">item_dict</span> <span class="o">=</span> <span class="p">{</span>
                <span class="s">&#39;chr&#39;</span><span class="p">:</span> <span class="n">chr_</span><span class="p">,</span>
                <span class="s">&#39;assembly&#39;</span><span class="p">:</span> <span class="n">assembly</span><span class="p">,</span>
                <span class="s">&#39;start&#39;</span><span class="p">:</span> <span class="nb">int</span><span class="p">(</span><span class="n">start</span><span class="p">),</span>
                <span class="s">&#39;end&#39;</span><span class="p">:</span> <span class="nb">int</span><span class="p">(</span><span class="n">end</span><span class="p">),</span>
                <span class="s">&#39;strand&#39;</span><span class="p">:</span> <span class="n">strand</span><span class="p">,</span>
            <span class="p">}</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">continue</span>
        <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">items</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s">&#39;;&#39;</span><span class="p">):</span>
            <span class="n">key</span><span class="p">,</span> <span class="n">value</span> <span class="o">=</span> <span class="n">item</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s">&#39;=&#39;</span><span class="p">)</span>
            <span class="k">if</span> <span class="n">key</span> <span class="o">==</span> <span class="s">&#39;Dbxref&#39;</span><span class="p">:</span>
                <span class="n">result_value</span> <span class="o">=</span> <span class="p">{}</span>
                <span class="k">for</span> <span class="n">subitem</span> <span class="ow">in</span> <span class="n">value</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s">&#39;,&#39;</span><span class="p">):</span>
                    <span class="n">subkey</span><span class="p">,</span> <span class="n">subvalue</span> <span class="o">=</span> <span class="n">subitem</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s">&#39;:&#39;</span><span class="p">)</span>
                    <span class="n">result_value</span><span class="p">[</span><span class="n">subkey</span><span class="p">]</span> <span class="o">=</span> <span class="n">subvalue</span>
                <span class="n">value</span> <span class="o">=</span> <span class="n">result_value</span>
            <span class="n">item_dict</span><span class="p">[</span><span class="n">key</span><span class="p">]</span> <span class="o">=</span> <span class="n">value</span>
        <span class="n">db</span><span class="p">[</span><span class="n">type_</span><span class="p">]</span><span class="o">.</span><span class="n">insert</span><span class="p">(</span><span class="n">item_list</span><span class="p">)</span>

<span class="n">db</span><span class="o">.</span><span class="n">exon_info</span><span class="o">.</span><span class="n">create_index</span><span class="p">([(</span><span class="s">&#39;transcript_id&#39;</span><span class="p">,</span> <span class="mi">1</span><span class="p">)])</span>
<span class="n">db</span><span class="o">.</span><span class="n">mRNA</span><span class="o">.</span><span class="n">create_index</span><span class="p">([(</span><span class="s">&#39;gene&#39;</span><span class="p">,</span> <span class="mi">1</span><span class="p">)])</span>
</pre></div>


<h2>连接数据库查询</h2>
<p>搭建数据库服务器的工作完成之后, 就需要进行本地windows环境的开发工作, 首先要实现的是内核部分的查询操作:</p>
<ol>
<li>首先, 我们需要根据基因名称, 列出所有的转录组编号;</li>
<li>根据上一步查询到的转录组信息, 列出转录组编号为上述转录组的所有外显子信息;</li>
</ol>
<p>得益于我们创建数据库时的索引方式, 上述查询非常简单快速, 其基本代码为:</p>
<div class="highlight"><pre><span class="kn">from</span> <span class="nn">pymongo</span> <span class="kn">import</span> <span class="n">MongoClient</span>
<span class="n">client</span> <span class="o">=</span> <span class="n">MongoClient</span><span class="p">(</span><span class="s">&#39;192.168.6.4&#39;</span><span class="p">,</span> <span class="mi">27017</span><span class="p">)</span>
<span class="n">db</span> <span class="o">=</span> <span class="n">client</span><span class="o">.</span><span class="n">reference</span>
<span class="n">db</span><span class="o">.</span><span class="n">authenticate</span><span class="p">(</span><span class="s">&#39;user&#39;</span><span class="p">,</span> <span class="s">&#39;123&#39;</span><span class="p">)</span>
<span class="n">mRNA_info</span> <span class="o">=</span> <span class="n">db</span><span class="o">.</span><span class="n">mRNA_info</span>
<span class="n">exon_info</span> <span class="o">=</span> <span class="n">db</span><span class="o">.</span><span class="n">exon_info</span>
<span class="n">gene_name</span> <span class="o">=</span> <span class="s">&#39;BRCA1&#39;</span>
<span class="n">cursor_mRNA</span> <span class="o">=</span> <span class="n">mRNA_info</span><span class="o">.</span><span class="n">find</span><span class="p">({</span><span class="s">&#39;gene&#39;</span><span class="p">:</span> <span class="n">gene_name</span><span class="p">},</span> <span class="p">{</span><span class="s">&#39;Name&#39;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> <span class="s">&#39;_id&#39;</span><span class="p">:</span> <span class="mi">0</span><span class="p">})</span>
<span class="k">for</span> <span class="n">q</span> <span class="ow">in</span> <span class="n">cursor_mRNA</span><span class="p">:</span>
    <span class="n">mRNA</span> <span class="o">=</span> <span class="n">q</span><span class="p">[</span><span class="s">&#39;Name&#39;</span><span class="p">]</span>
    <span class="n">exon_length</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="n">exon_num</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="n">cursor_exon</span> <span class="o">=</span> <span class="n">exon_info</span><span class="o">.</span><span class="n">find</span><span class="p">({</span><span class="s">&#39;transcript_id&#39;</span><span class="p">:</span> <span class="n">mRNA</span><span class="p">},</span> <span class="p">{</span><span class="s">&#39;start&#39;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> <span class="s">&#39;end&#39;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> <span class="s">&#39;_id&#39;</span><span class="p">:</span> <span class="mi">0</span><span class="p">})</span>
    <span class="k">for</span> <span class="n">q</span> <span class="ow">in</span> <span class="n">cursor_exon</span><span class="p">:</span>
        <span class="n">exon_length</span> <span class="o">+=</span> <span class="n">q</span><span class="p">[</span><span class="s">&#39;end&#39;</span><span class="p">]</span> <span class="o">-</span> <span class="n">q</span><span class="p">[</span><span class="s">&#39;start&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1</span>
        <span class="n">exon_num</span> <span class="o">+=</span> <span class="mi">1</span>
     <span class="k">print</span><span class="p">(</span><span class="n">gene_name</span><span class="p">,</span> <span class="n">mRNA</span><span class="p">,</span> <span class="n">exon_length</span><span class="p">,</span> <span class="n">exon_num</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&#39;</span><span class="se">\t</span><span class="s">&#39;</span><span class="p">)</span>
</pre></div>


<h2>PyQt设计界面</h2>
<p>至此, 核心任务已经完成, 但是达到文首的目的, 所以还需要再设计一个桌面应用, 我选择使用PyQt5来完成GUI部分的工作;</p>
<p>首先, 利用QtDesigner, 设计一个粗糙简单的界面:</p>
<p><img alt="QtDesigner设计结果" src="http://ww3.sinaimg.cn/large/95202659gw1f4dffbs4vmj21hc0sx0wv.jpg" /></p>
<p>由于本人艺术细胞缺乏, 就只弄一个粗糙的界面了;</p>
<p>保存为: main.ui, 再使用以下命令将其转换为ui模块:</p>
<div class="highlight"><pre><span class="n">python</span> <span class="o">-</span><span class="n">m</span> <span class="n">PyQt5</span><span class="o">.</span><span class="n">uic</span><span class="o">.</span><span class="n">pyuic</span> <span class="n">main</span><span class="o">.</span><span class="n">ui</span> <span class="o">-</span><span class="n">o</span> <span class="n">my_design</span><span class="o">.</span><span class="n">py</span>
</pre></div>


<p>生成的my_design.py中就包含了我们设计的界面信息;</p>
<h2>添加界面逻辑</h2>
<p>接下来要做的工作就是将界面和核心代码融合, 首先我们要确定界面中信号的传递方式:</p>
<ol>
<li>首先通过按钮来获得查询信号;</li>
<li>然后从输入栏中获得基因名称;</li>
<li>再利用该基因名称获取查询结果并格式化为字符串;</li>
<li>将字符串展示到展示框中;</li>
</ol>
<p>详细代码为:</p>
<div class="highlight"><pre><span class="kn">import</span> <span class="nn">sys</span>
<span class="kn">from</span> <span class="nn">pymongo</span> <span class="kn">import</span> <span class="n">MongoClient</span>
<span class="kn">from</span> <span class="nn">PyQt5.QtWidgets</span> <span class="kn">import</span> <span class="o">*</span>
<span class="kn">from</span> <span class="nn">my_design</span> <span class="kn">import</span> <span class="n">Ui_MainWindow</span>

<span class="n">client</span> <span class="o">=</span> <span class="n">MongoClient</span><span class="p">(</span><span class="s">&#39;192.168.6.4&#39;</span><span class="p">,</span> <span class="mi">27017</span><span class="p">)</span>
<span class="n">db</span> <span class="o">=</span> <span class="n">client</span><span class="o">.</span><span class="n">reference</span>
<span class="n">db</span><span class="o">.</span><span class="n">authenticate</span><span class="p">(</span><span class="s">&#39;user&#39;</span><span class="p">,</span> <span class="s">&#39;123&#39;</span><span class="p">)</span>
<span class="n">mRNA_info</span> <span class="o">=</span> <span class="n">db</span><span class="o">.</span><span class="n">mRNA_info</span>
<span class="n">exon_info</span> <span class="o">=</span> <span class="n">db</span><span class="o">.</span><span class="n">exon_info</span>


<span class="k">class</span> <span class="nc">MyWindow</span><span class="p">(</span><span class="n">QMainWindow</span><span class="p">,</span> <span class="n">Ui_MainWindow</span><span class="p">):</span>
    <span class="k">def</span> <span class="nf">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="nb">super</span><span class="p">()</span><span class="o">.</span><span class="n">__init__</span><span class="p">()</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">setupUi</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">pushButton</span><span class="o">.</span><span class="n">clicked</span><span class="o">.</span><span class="n">connect</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">show_search_result</span><span class="p">)</span>

    <span class="k">def</span> <span class="nf">show_search_result</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="n">gene_name</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">lineEdit</span><span class="o">.</span><span class="n">text</span><span class="p">()</span>
        <span class="n">search_result</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">search</span><span class="p">(</span><span class="n">gene_name</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">textBrowser</span><span class="o">.</span><span class="n">setText</span><span class="p">(</span><span class="n">search_result</span><span class="p">)</span>

    <span class="k">def</span> <span class="nf">search</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">gene_name</span><span class="p">):</span>
        <span class="n">result</span> <span class="o">=</span> <span class="s">&#39;&#39;</span>
        <span class="n">cursor_mRNA</span> <span class="o">=</span> <span class="n">mRNA_info</span><span class="o">.</span><span class="n">find</span><span class="p">({</span><span class="s">&#39;gene&#39;</span><span class="p">:</span> <span class="n">gene_name</span><span class="p">},</span> <span class="p">{</span><span class="s">&#39;Name&#39;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> <span class="s">&#39;_id&#39;</span><span class="p">:</span> <span class="mi">0</span><span class="p">})</span>
        <span class="k">for</span> <span class="n">q</span> <span class="ow">in</span> <span class="n">cursor_mRNA</span><span class="p">:</span>
            <span class="n">mRNA</span> <span class="o">=</span> <span class="n">q</span><span class="p">[</span><span class="s">&#39;Name&#39;</span><span class="p">]</span>
            <span class="n">exon_length</span> <span class="o">=</span> <span class="mi">0</span>
            <span class="n">exon_num</span> <span class="o">=</span> <span class="mi">0</span>
            <span class="n">cursor_exon</span> <span class="o">=</span> <span class="n">exon_info</span><span class="o">.</span><span class="n">find</span><span class="p">({</span><span class="s">&#39;transcript_id&#39;</span><span class="p">:</span> <span class="n">mRNA</span><span class="p">},</span> 
                                         <span class="p">{</span><span class="s">&#39;start&#39;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> <span class="s">&#39;end&#39;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> <span class="s">&#39;_id&#39;</span><span class="p">:</span> <span class="mi">0</span><span class="p">})</span>
            <span class="k">for</span> <span class="n">q</span> <span class="ow">in</span> <span class="n">cursor_exon</span><span class="p">:</span>
                <span class="n">exon_length</span> <span class="o">+=</span> <span class="n">q</span><span class="p">[</span><span class="s">&#39;end&#39;</span><span class="p">]</span> <span class="o">-</span> <span class="n">q</span><span class="p">[</span><span class="s">&#39;start&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1</span>
                <span class="n">exon_num</span> <span class="o">+=</span> <span class="mi">1</span>
            <span class="n">result</span> <span class="o">+=</span> <span class="s">&#39;{gene_name}</span><span class="se">\t</span><span class="s">{mRNA}</span><span class="se">\t</span><span class="s">{exon_length}</span><span class="se">\t</span><span class="s">{exon_num}</span><span class="se">\n</span><span class="s">&#39;</span><span class="o">.</span><span class="n">format_map</span><span class="p">(</span><span class="nb">vars</span><span class="p">())</span>
        <span class="k">return</span> <span class="n">result</span>


<span class="k">if</span> <span class="n">__name__</span> <span class="o">==</span> <span class="s">&#39;__main__&#39;</span><span class="p">:</span>
    <span class="n">app</span> <span class="o">=</span> <span class="n">QApplication</span><span class="p">(</span><span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">)</span>
    <span class="n">window</span> <span class="o">=</span> <span class="n">MyWindow</span><span class="p">()</span>
    <span class="n">window</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
    <span class="n">sys</span><span class="o">.</span><span class="n">exit</span><span class="p">(</span><span class="n">app</span><span class="o">.</span><span class="n">exec_</span><span class="p">())</span>
</pre></div>


<h2>打包发布</h2>
<p>到这里基本工作已经完成, 但是目前只是在开发环境下运行, 如果要release到其他机器上, 需要对依赖的库进行打包;</p>
<p>使用<strong>pyinstaller</strong>工具可以非常方便地完成这项工作;</p>
<ol>
<li>首先创建一个打包后的目标目录, 如: app;</li>
<li>进入该目录, 运行: pyinstaller -w script_name.py; (主程序文件替换为完整目录)</li>
<li>等待5分钟左右, 即可在该目录下看到两个文件夹: build、dist, 其中dist目录下的exe文件即最终打包好的文件;</li>
</ol>
<p>然而普通的电脑上运行这个exe文件还是会有报错: </p>
<div class="highlight"><pre>could not be located in the dynamic link library
api-ms-win-crt-runtime-|1-1-0.dll
</pre></div>


<p>这是因为运行的机器上没有安装vc++ 2015的依赖库, 去以下地址下载并且安装即可: </p>
<ul>
<li><a href="http://download.microsoft.com/download/9/3/F/93FCF1E7-E6A4-478B-96E7-D4B285925B00/vc_redist.x64.exe">64位下载地址</a></li>
<li><a href="http://download.microsoft.com/download/9/3/F/93FCF1E7-E6A4-478B-96E7-D4B285925B00/vc_redist.x86.exe">32位下载地址</a></li>
</ul>
<p>双击运行, 并输入BRCA1基因测试:</p>
<p><img alt="BRCA1 result" src="http://ww2.sinaimg.cn/large/95202659gw1f4dfx7y7nfj20m90hj0sz.jpg" /></p>
<h2>后记</h2>
<p>至此所有工作已经完成了, 但是这个软件并不完美(不仅仅是界面), 后续可以往上面添加更多的内容, 如异常处理, 通过基因别名查询, 更多的查询方式等等功能;</p>
            </div>
            <!-- /.entry-content -->

    <hr/>
    <section class="comments" id="comments">
        <h2>Comments</h2>
<!--         <div class="ds-thread" data-thread-key="di-yi-ge-sheng-wu-xin-xi-app" data-title="第一个生物信息app" data-url="http://www.lyon0804.com/di-yi-ge-sheng-wu-xin-xi-app.html"></div> -->
		<script type="text/javascript">
     		var duoshuoQuery = {short_name:"lyon0804"};
    		(function() {
    			var ds = document.createElement('script');
    			ds.type = 'text/javascript';ds.async = true;
    			ds.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') + '//static.duoshuo.com/embed.js';
    			ds.charset = 'UTF-8';
    			(document.getElementsByTagName('head')[0]
    			 || document.getElementsByTagName('body')[0]).appendChild(ds);
    		})();
    	</script>
        <script type="text/javascript">
            function toggleDuoshuoComments(container, id, title, url){
                if(jQuery(container).has("div").length>0){
                    jQuery(container).empty();
                    return;
                }
                var el = document.createElement('div');
                el.setAttribute('class','ds-thread');
                el.setAttribute('data-thread-key', id);
                el.setAttribute('data-title',title);
                el.setAttribute('data-url', url);
                DUOSHUO.EmbedThread(el);
                jQuery(container).append(el);
            }
        </script>
        <a href="javascript:void(0);" onclick="toggleDuoshuoComments('#comment-box', 'di-yi-ge-sheng-wu-xin-xi-app', '第一个生物信息app' , 'http://www.lyon0804.com/di-yi-ge-sheng-wu-xin-xi-app.html');">
        查看评论</a>
        <div id="comment-box" ></div>
        <hr/>
    	<noscript>Please enable JavaScript to view the <a href="http://duoshuo.com/">comments powered by
        Duoshuo.</a></noscript>
    	<a href="http://duoshuo.com" class="dsq-brlink">comments powered by <span>Duoshuo</span></a>

    </section>
        </article>
    </section>

        </div>
    </div>
</div>
<footer>
   <div class="container">
      <hr>
      <div class="row">
            <!-- <div class="list-group-item"> -->
            <!-- <h4> -->
            <!-- <i class="fa fa-home fa-lg"></i> -->
            <!-- <span class="icon-label">Social</span></h4> -->
              <ul class="list-group" id="social">
                   <!-- <li class="list-group-item"> -->
                   <div class="social-item">
                   <a href="http://github.com/lozybean">
                     <i class="fa fa-github-square fa-lg"></i><!--  github -->
                   </a>
                   </div>
                   <!-- </li> -->
                   <!-- <li class="list-group-item"> -->
                   <div class="social-item">
                   <a href="http://weibo.com/lyon0804">
                     <i class="fa fa-weibo fa-lg"></i><!--  weibo -->
                   </a>
                   </div>
                   <!-- </li> -->
              </ul>
            <!-- </div> -->
         <div class="col-xs-10">&copy; 2016 lyon
            &middot; Powered by <a href="https://github.com/DandyDev/pelican-bootstrap3" target="_blank">pelican-bootstrap3(modified by lyon)</a>,
            <a href="http://docs.getpelican.com/" target="_blank">Pelican</a>,
            <a href="http://getbootstrap.com" target="_blank">Bootstrap</a>         </div>
         <div class="col-xs-2" id="back-top">
            <!-- <p class="pull-right"><i class="fa fa-arrow-up"></i>  -->
            <a href="#top" title="回到顶部"></a>
            </p>
         </div>
      </div>
   </div>
</footer>
<script src="http://www.lyon0804.com/theme/js/jquery.min.js"></script>

<!-- Include all compiled plugins (below), or include individual files as needed -->
<script src="http://www.lyon0804.com/theme/js/bootstrap.min.js"></script>

<!-- Enable responsive features in IE8 with Respond.js (https://github.com/scottjehl/Respond) -->
<script src="http://www.lyon0804.com/theme/js/respond.min.js"></script>


</body>
</html>